lib/JsonNode.cpp

```diff
@@ -872,23 +874,42 @@ void JsonUtils::merge(JsonNode & dest, JsonNode & source)
 		}
 		case JsonNode::JsonType::DATA_STRUCT:
 		{
-			//recursively merge all entries from struct
-			for(auto & node : source.Struct())
-				merge(dest[node.first], node.second);
+			bool override = false;
+			if(!noOverride)
+			{
+				for(std::string flag : source.flags)
+				{
+					if(flag.compare("override") == 0)
+					{
+						override = true;
+						break;
+					}
+				}
+			}
+			if(override)

```

Comment:
Construction above seems like overkill. This should work:
```
if(!noOverride && vstd::contains(source.flags, "override"))
```
I understand that you want to make code more explicit, but I really doubt it's required there.

Reply:
Just been too long that I coded much for VCMI, forgot we had that handy function. Fixed.